src/ServiceInsight/Search/SearchBarView.xaml

```diff
@@ -24,7 +24,7 @@
                 Padding="5"
                 HorizontalContentAlignment="Center"
                 VerticalContentAlignment="Center"
-                cal:Message.Attach="[Event Click]=[Action Search]">
+                cal:Message.Attach="[Event Click]=[Action RefreshResult]">
```

Comment:
The search public method is still there. The RefreshResult is directly calling Search, so why not bind it directly to Search and remove the extra method?

Reply:
Because `CanSearch` is already defined and means something else. I had to use an explicit different name here so that I can have proper `Can*` property.

Reply:
I still struggle to see the issue. If `CanSearch` is already defined that's fine too, since that evaluates the execution of both Search/RefreshResult (since that's what RefreshResult is calling anyway?)

Maybe jump on a call when you're feeling better and are back to work?

Reply:
@HEskandari the `CanSearch` is used in the search button as is defined as

```
public bool CanSearch => !WorkInProgress && !string.IsNullOrWhiteSpace(SearchQuery);
```

Which is different from `CanRefreshResult`:

```
public bool CanRefreshResult => !WorkInProgress;
```

The behavior of both buttons is the same but the conditions under which they are enabled are different. That's why I added the seemingly not required separate method.